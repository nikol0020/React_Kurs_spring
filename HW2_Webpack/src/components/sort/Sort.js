import React, {Component} from 'react';
import './Sort.scss';
import { movieShape } from '../propTypes';
import PropTypes from 'prop-types';

import Button from '../button';

export default class Sort extends Component {
    render() {
        return (
            <div className="sort">
                <span className="sort__text">Sort by</span>
                <Button className="button" value="rel-s date"/>
                <Button className="button" value="rating"/>
            </div>
        )
    }
}

Sort.propTypes = {
    data: PropTypes.arrayOf(movieShape)
};