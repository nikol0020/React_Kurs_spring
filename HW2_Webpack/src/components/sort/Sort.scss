.sort {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 300px;
  height: 60px;
  margin-top: 20px;
  margin-left: 50px;

  &__text, &__buttons {
     z-index: 1;
   }

  &__text {
    z-index: 1;
    color: #888;
  }

  &__buttons {
    padding: 7px 15px;
    border-radius: 5px;

    &:hover {
      cursor: pointer;
    }
  }
}