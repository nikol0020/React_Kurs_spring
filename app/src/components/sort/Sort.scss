.sort {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 300px;
  height: 50px;
  margin-top: 20px;
  margin-left: 50px;

  &__text {
    z-index: 1;
  }

  &__buttons {
    padding: 7px 15px;
    border-radius: 5px;
  }
}