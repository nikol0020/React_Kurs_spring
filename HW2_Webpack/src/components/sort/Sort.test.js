import React from 'react';
import Sort from './Sort';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('<Sort /> shallow rendering', () => {

    it('matches the snapshot Sort', () => {
        const wrapper = shallow(<Sort />);
        expect(wrapper).toMatchSnapshot();
    });

    it('<span>Sort by</span>', () => {
        const wrapper = shallow(<Sort />);
        expect(wrapper.find('span').text()).toBe('Sort by');
    });

    it('should render a div', () => {
        const wrapper = shallow(
            <div className="sort"></div>
        );
        expect(wrapper).toMatchSnapshot();
    });
});