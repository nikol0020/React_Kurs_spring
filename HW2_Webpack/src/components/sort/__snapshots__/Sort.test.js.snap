// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort /> shallow rendering matches the snapshot Sort 1`] = `
<div
  className="sort"
>
  <span
    className="sort__text"
  >
    Sort by
  </span>
  <Button
    className="button"
    value="rel-s date"
  />
  <Button
    className="button"
    value="rating"
  />
</div>
`;

exports[`<Sort /> shallow rendering should render a div 1`] = `
<div
  className="sort"
/>
`;
